pyanaconda/ui/tui/spokes/askvnc.py

```diff
@@ -105,7 +105,12 @@ def input(self, args, key):
             d = YesNoDialog(self.app, _(self.app.quit_message))
             self.app.switch_screen_modal(d)
             if d.answer:
-                execWithRedirect("systemctl", ["--no-wall", "reboot"])
+                from pyanaconda.flags import flags
+                if not flags.imageInstall and not flags.livecdInstall \
+                   and not flags.dirInstall:
```

Comment:
Should use ``can_touch_runtime_system`` instead. That would also catch the ``flags.testing`` case.